src/PlotDialog.cpp

```
@@ -201,30 +207,41 @@ void PlotDialog::GetScale() {
     bool first = true;
     for (std::list<PlotData>::iterator it = m_PlotData.begin();
          it != m_PlotData.end(); it++) {
-      double value = GetValue(*it, cVariable[i]->GetSelection());
+      Variable variable =
+          GetVariableEnumFromIndex(cVariable[i]->GetSelection());
+      double value = GetValue(*it, variable);
       if (first) {
         m_StartTime = (*it).time;
         m_mintime = m_maxtime = 0.;
         m_minvalue[i] = m_maxvalue[i] = value;
         first = false;
       } else {
-        double time = ((*it).time - m_StartTime).GetSeconds().ToDouble();
-        m_mintime = MIN(time, m_mintime);
-        m_maxtime = MAX(time, m_maxtime);
+        // Make sure both datetimes are valid before performing subtraction
+        if ((*it).time.IsValid() && m_StartTime.IsValid()) {
+          double time = ((*it).time - m_StartTime).GetSeconds().ToDouble();
+          m_mintime = MIN(time, m_mintime);
+          m_maxtime = MAX(time, m_maxtime);
+        } else {
+          // Skip this point if either datetime is invalid
```

Would it be better to generate an error here, rather than silently ignore the invalid data?